### PR TITLE
Improve Pulse docs: "What is Pulse?" and "Getting started" 

### PR DESCRIPTION
**Pull request by nikolasburk**
I made a couple of changes that I hope make it easier for people to understand what Pulse is, what value it provides and getting started with it.

Nothing too fundamental, just a few things I thought should be improved:

- What is Pulse?
  - Added more structure/headings to the page
  - Laid out use case and features (based on the value messaging guide)
  - Update existing and add new illustrations (design tasks are [here](https://www.notion.so/prismaio/Pulse-illustrations-fcb09e704c5d4435904e8d5da09e11cb?pvs=4))
  - Adjusted and expanded on some of the existing explanations
- Getting started
  - Clean up prerequisites 
  - Divide step 1. into smaller steps to handhold the reader more
  - Change `prisma.notification.subscribe()` to `prisma.user.subscribe()` and add the corresponding Prisma schema in a tab
  - add step 3. to test the subscription
  - updated "Next steps" with more filter examples

FWIW: I didn't do that yet but I think we should also (in the future) instruct people to run `prisma migrate` in these instructions to really have a fully step-by-step tutorial that doesn't leave room for error

### CODE REVIEW

File: content/400-pulse/100-what-is-pulse.mdx [hunk 1]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
 ---
 
-<TopBlock>
+## What is Prisma Pulse?
 
-[Prisma Pulse](https://www.prisma.io/data-platform/pulse) is managed database-event infrastructure that captures and distributes your database events to your application. It simplifies subscribing to type-safe data changes with an extended [Prisma Client](/orm/prisma-client) to power real-time functionality.
+[Prisma Pulse](https://www.prisma.io/data-platform/pulse) is managed infrastructure that captures databases events (i.e. create, update and delete operations) and distributes them to your applications.
 
-</TopBlock>
+It lets you subsribe to any changes happening in your database using [Prisma Client](/orm/prisma-client) to power **real-time use cases** in your applications, such as:
```
Comment (jharrell) [inline content/400-pulse/100-what-is-pulse.mdx]:
```suggestion
It lets you subscribe to any changes happening in your database using [Prisma Client](/orm/prisma-client) to power **real-time use cases** in your applications, such as:
```

File: content/400-pulse/100-what-is-pulse.mdx [hunk 2]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
+- chat and messaging
+- data syncing (e.g. into a search index)
+- social interaction and collaboration
+- notifying users
+- updating inventories
+- managing payments
+- ... and any other functionality that requires real-time updates
 
-Prisma Pulse leverages Change Data Capture (CDC) to efficiently observe and capture database changes as they occur. By monitoring the database's transaction log, Prisma Pulse identifies change events like inserts, updates, and deletes without impacting the database's performance.
+Prisma Pulse integrates with [Prisma ORM](/orm) and easily lets you subscribe database changes in a type-safe way using Prisma Client:
```
Comment (jharrell) [inline content/400-pulse/100-what-is-pulse.mdx]:
```suggestion
Prisma Pulse integrates with [Prisma ORM](/orm) and lets you subscribe to database changes easily _and_ in a type-safe way using Prisma Client:
```

File: content/400-pulse/100-what-is-pulse.mdx [hunk 3]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
+![](/img/pulse/overview.png)
 
-This eliminates the need for complex polling or manual data synchronization, saving you development time and effort.
+## Why Prisma Pulse?
 
-![What is Pulse](/img/pulse/what-is-pulse.png)
+Building real-time functionality based on changes that occur in your database can be very complicated. Common approaches like _polling_, implementing _application-level updates_ or using _additional infrastructure_ (like Apache Kafka or RabbitMQ) either don't scale, are very costly or come with substantial development and maintenance overhead.
 
-## What you can build with Prisma Pulse
+Prisma Pulse is based on the idea of _unidirectional data flow_ which is implemented via [Change Data Capture](https://en.wikipedia.org/wiki/Change_data_capture) (CDC). CDC gets rid of the deficincies of the other approaches and solves the problem of reacting to database events in a robust and elegant manner.
```
Comment (jharrell) [inline content/400-pulse/100-what-is-pulse.mdx]: When we say “other approaches” what do we mean here?

File: content/400-pulse/100-what-is-pulse.mdx [hunk 4]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
+- Reacting to database changes with type-safe model subscriptions
+- Unidirectional data flow via Change Data Capture
+- Great DX integrated with usage of Prisma ORM (easy setup, development and maintenance)
+- Works with your existing database
+- Insights dashboard lets you view and understand all database events captured by Pulse
+- Enable, disable and manage Pulse via the [Platform CLI](/platform/platform-cli/)
+
+## How does Prisma Pulse work?
+
+Prisma Pulse uses CDC to efficiently observe and capture database changes as they occur. By monitoring the database's transaction log (e.g. the Write-Ahead-Log in PostgreSQL), Prisma Pulse identifies change events like inserts, updates, and deletes without impacting the database's performance.
```
Comment (jharrell) [inline content/400-pulse/100-what-is-pulse.mdx]: Is transaction log the best name? I thought it was called the replication log? (I could be wrong)

**Review by jharrell — CHANGES_REQUESTED** [review]
Few minor changes and some questions

File: content/400-pulse/100-what-is-pulse.mdx [hunk 5]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
-[Prisma Pulse](https://www.prisma.io/data-platform/pulse) is managed database-event infrastructure that captures and distributes your database events to your application. It simplifies subscribing to type-safe data changes with an extended [Prisma Client](/orm/prisma-client) to power real-time functionality.
+[Prisma Pulse](https://www.prisma.io/data-platform/pulse) is managed infrastructure that captures databases events (i.e. create, update and delete operations) and distributes them to your applications.
 
-</TopBlock>
+It lets you subsribe to any changes happening in your database using [Prisma Client](/orm/prisma-client) to power **real-time use cases** in your applications, such as:
 
-## How Prisma Pulse works
+- chat and messaging
+- data syncing (e.g. into a search index)
+- social interaction and collaboration
```
Comment (petradonka) [inline content/400-pulse/100-what-is-pulse.mdx]: What is "social interaction"? How about roughly syncing up this list with what's on prisma.io/pulse? It's already very close :)

File: content/400-pulse/100-what-is-pulse.mdx [hunk 6]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
+- notifying users
+- updating inventories
+- managing payments
+- ... and any other functionality that requires real-time updates
 
-Prisma Pulse leverages Change Data Capture (CDC) to efficiently observe and capture database changes as they occur. By monitoring the database's transaction log, Prisma Pulse identifies change events like inserts, updates, and deletes without impacting the database's performance.
+Prisma Pulse integrates with [Prisma ORM](/orm) and easily lets you subscribe database changes in a type-safe way using Prisma Client:
 
-The captured events are processed, evaluated, and swiftly distributed to relevant client subscriptions ensuring your applications stay synchronized with the latest database state.
+![](/img/pulse/overview.png)
```
Comment (petradonka) [inline content/400-pulse/100-what-is-pulse.mdx]: I think a lot of this image could instead be an ordered list and code snippets, plus some smaller parts of the visual, which would flow better with the docs page. What do you think?

File: content/400-pulse/100-what-is-pulse.mdx (hunk 1 above)
Comment (petradonka) [inline content/400-pulse/100-what-is-pulse.mdx]: Could we also mention event-driven workflows or architectures? Either here, or another section if you find it more suitable.

File: content/400-pulse/100-what-is-pulse.mdx [hunk 7]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
-This eliminates the need for complex polling or manual data synchronization, saving you development time and effort.
+## Why Prisma Pulse?
 
-![What is Pulse](/img/pulse/what-is-pulse.png)
+Building real-time functionality based on changes that occur in your database can be very complicated. Common approaches like _polling_, implementing _application-level updates_ or using _additional infrastructure_ (like Apache Kafka or RabbitMQ) either don't scale, are very costly or come with substantial development and maintenance overhead.
 
-## What you can build with Prisma Pulse
+Prisma Pulse is based on the idea of _unidirectional data flow_ which is implemented via [Change Data Capture](https://en.wikipedia.org/wiki/Change_data_capture) (CDC). CDC gets rid of the deficincies of the other approaches and solves the problem of reacting to database events in a robust and elegant manner.
+
+Here is an overview of the main features Prisma Pulse provides:
```
Comment (petradonka) [inline content/400-pulse/100-what-is-pulse.mdx:76]: This is great!

File: content/400-pulse/100-what-is-pulse.mdx (hunk 4 above)
Comment (petradonka) [inline content/400-pulse/100-what-is-pulse.mdx]:
```suggestion
Prisma Pulse uses CDC to efficiently observe and capture database changes as they occur. By monitoring the database's transaction log (e.g. the Write-Ahead-Log in PostgreSQL), Prisma Pulse identifies database change events like inserts, updates, and deletes without impacting the database's performance.
```

Added for more clarity & consistency, as this is an introductory section and it's good to be a bit extra verbose in terminology.

File: content/400-pulse/100-what-is-pulse.mdx [hunk 8]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
+- Insights dashboard lets you view and understand all database events captured by Pulse
+- Enable, disable and manage Pulse via the [Platform CLI](/platform/platform-cli/)
+
+## How does Prisma Pulse work?
+
+Prisma Pulse uses CDC to efficiently observe and capture database changes as they occur. By monitoring the database's transaction log (e.g. the Write-Ahead-Log in PostgreSQL), Prisma Pulse identifies change events like inserts, updates, and deletes without impacting the database's performance.
+
+The captured events are processed, evaluated, and distributed to all Prisma Client instances that subscribed to them:
+
+![What is Pulse](/img/pulse/unidirectional-data-flow.png)
```
Comment (petradonka) [inline content/400-pulse/100-what-is-pulse.mdx:91]: Will we add Alejandro-versions of these visuals when he's back? :)

File: content/400-pulse/100-what-is-pulse.mdx [hunk 9]
```diff
@@ -6,23 +6,48 @@ tocDepth: 3
...
+
+## What can you build with Prisma Pulse?
```
Comment (petradonka) [inline content/400-pulse/100-what-is-pulse.mdx:93]: I really wish GitHub would allow me to comment on lines not changed. Anyway: could we rename the header "Examples" to "Example projects", to be more specific? I'd be more excited to look at example _projects_ than "examples", where maybe I find snippets, maybe just text, who knows, but "Example projects" is definitely code, and more exciting.

File: content/400-pulse/200-getting-started.mdx [hunk 10]
```diff
@@ -17,19 +16,39 @@ Prisma Pulse currently supports PostgreSQL. We'd love to hear [which databases](
 
 To get started with Pulse, you will need the following:
 
-- A [Prisma Data Platform workspace](https://console.prisma.io).
-- [Prisma Client](/orm/prisma-client) version `4.16.1` or higher and [`@prisma/extension-pulse`](https://www.npmjs.com/package/@prisma/extension-pulse) version `1.0.1` or higher.
-- A publicly accessible PostgreSQL (version 12+) database with [logical replication](https://www.postgresql.org/docs/current/logical-replication-quick-setup.html) enabled. View our [setup guide](/pulse/database-setup/general-database-instructions#enable-logical-replication) on configuring logical replication for your database.
+- An account in the [Prisma Data Platform](https://console.prisma.io).
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx]:
```suggestion
- A [Prisma Data Platform](https://console.prisma.io) account
```

File: content/400-pulse/200-getting-started.mdx [hunk 11]
```diff
@@ -17,19 +16,39 @@ Prisma Pulse currently supports PostgreSQL. We'd love to hear [which databases](
...
+- The **connection string** of a Pulse-ready database (if you don't have one yet, you can configure your database the instructions [here](/pulse/database-setup) or [use a Railway template](https://railway.app/template/pulse-pg?referralCode=VQ09uv))
+
+## 1. Enable Pulse in the Platform Console
+
+### 1.1. Select workspace -> project -> environment
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx]: I really like that you spell this out. How about this, considering the additional notes you have below:

```suggestion
### 1.1. Choose the environment you want to enable Pulse for
```

And then below, maybe also worth noting that if you want (or have no projects yet), you can create a new project?

File: content/400-pulse/200-getting-started.mdx [hunk 12]
```diff
@@ -17,19 +16,39 @@ Prisma Pulse currently supports PostgreSQL. We'd love to hear [which databases](
...
+
+Open the [Platform Console](https://console.prisma.io), navigate to your workspace of choice, then select the project and choose the environment in which you want to enable Pulse.
+
+### 1.2. Enable Pulse
+
+When you're there, click the **Enable Pulse** button.
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx]:
```suggestion
In the project environment of your choice, click the **Enable Pulse** button.
```

File: content/400-pulse/200-getting-started.mdx [hunk 13]
```diff
@@ -17,19 +16,39 @@ Prisma Pulse currently supports PostgreSQL. We'd love to hear [which databases](
...
+The Pulse **Setup** screen requires you to:
 
-> Once enabled, you'll be prompted to generate an API key that you'll use in your extended Prisma Client to authenticate requests. Store this API key in your application's `.env` file:
->
-> ```env file=.env showLineNumbers
-> PULSE_API_KEY="your_secure_pulse_api_key"
-> ```
+- provide your **Database connection string**
+- select a **Region** where Pulse should be hosted
+- decide whether you want to use the **Automatic setup** for **Database replication** (only available on paid plans)
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx]: What is automatic setup? Can we link somewhere to read more about that? (Or add some additional details?)

File: content/400-pulse/200-getting-started.mdx [hunk 14]
```diff
@@ -17,19 +16,39 @@ Prisma Pulse currently supports PostgreSQL. We'd love to hear [which databases](
...
+
+When you're done with that, click the **Enable Pulse** button at the bottom of the screen. This will test the connectivity to your database.
+
+
+### 1.4. Generate an API key
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx:44]: Do you need to generate a new API key, even if you already have one for this environment?

File: content/400-pulse/200-getting-started.mdx [hunk 15]
```diff
@@ -62,46 +81,91 @@ const prisma = new PrismaClient().$extends(
...
-In the below example, a subscription is made to a `notification` model that listens for _any_ change event on that table:
+In the below example, it is assumed that your Prisma schema has a `User` model. A subscription is made to that `User` model that listens for _any_ change event on that table:
+
+<TabbedContent code>
+<TabItem value="Subscription">
 
 ```ts
-const prisma = new PrismaClient().$extends(withPulse({ apiKey: apiKey }))
+import { PrismaClient } from '@prisma/client'
+import { withPulse } from '@prisma/extension-pulse'
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx:102]: Can we mark the added/changed lines with color as in git diffs?

File: content/400-pulse/200-getting-started.mdx [hunk 16]
```diff
@@ -62,46 +81,91 @@ const prisma = new PrismaClient().$extends(
...
+
+</TabbedContent>
+
+## 3. Test your subscription
+
+After running the code snippet above, you can test the subscription by _creating_, _updating_ or _deleting_ a `User` record in your database. 
+
+You can do that using Prisma Studio (by running `npx prisma studio`) or by using any other database client of your choice (like [Postico](https://eggerapps.at/postico2/) or `psql`).
+
+If everything worked, you should see the event being logged to the terminal where you can the code snippet from above.
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx]:
```suggestion
If everything worked, you should see the event being logged to the terminal where you can the code snippet from above. 🎉
```

:)

File: content/400-pulse/200-getting-started.mdx [hunk 17]
```diff
@@ -62,46 +81,91 @@ const prisma = new PrismaClient().$extends(
...
 
 ## Next steps
 
-[Navigate to the API section](/pulse/api-reference) to explore available filtering options for Pulse's `subscribe()` method.
+Yopu can try out more filters on your Pulse subscription, for example:
```
Comment (petradonka) [inline content/400-pulse/200-getting-started.mdx]:
```suggestion
You can try out more filters on your Pulse subscription, for example:
```

**Review by petradonka — CHANGES_REQUESTED** [review]
@nikolasburk, this is great, thanks so much for adding these details! 🙏

File: content/400-pulse/100-what-is-pulse.mdx (hunk 5 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx]: > What is "social interaction"?

I was thinking about "like","follow", etc features that you often find in social media apps. Do you feel it's not clear enough and I should drop that altogether?

> How about roughly syncing up this list with what's on prisma.io/pulse? It's already very close :)

I did take inspiration from that list in the first place and took all the use case that I felt would work in this bullet list. I just looked at it again and added "onboarding emails" as another use case here: 

- chat and messaging
- sending onboarding emails
- data syncing (e.g. into a search index)
- social interaction and collaboration
- notifying users
- updating inventories
- managing payments
- ... and any other functionality that requires real-time updates

Are there any others you think should be added?

File: content/400-pulse/100-what-is-pulse.mdx (hunk 6 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx]: Hmm, I prefer the power of a good visual (of which I think we generally have way too little in the docs).

But I see how this can be turned into text-based content where essentially only the illustration of step 2. is needed because everything else is text and code snippets?

I would probably still not use a bullet list but rather use numbered headings like:

### 1. Subscribe

Subscribe to changes on a model with Prisma Client, e.g. `User`.

```
// subscribe code
```

### 2. Database event jhappens

A change is made ...



### 3. Database event is propagated

Prisma Pulse propages ....

```
// terminal output
```

--- 

Do you prefer that remove the illustration that depicts the whole flow and turn it into this text-based version?

File: content/400-pulse/100-what-is-pulse.mdx (hunk 3 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx]: It's referencing the "Common approaches like _polling_, implementing _application-level updates_ or using _additional infrastructure_ (like Apache Kafka or RabbitMQ) either don't scale, are very costly or come with substantial development and maintenance overhead" from the first paragraph.

If that wasn't clear to you while reading, do you have an idea how this can be made clearer?

File: content/400-pulse/100-what-is-pulse.mdx (hunk 8 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx:91]: Of course! I have already created the design tasks but still find that it's good practice to include the draft illustrations so that reviewers can know roughly what to expect from a visual!

File: content/400-pulse/100-what-is-pulse.mdx (hunk 4 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx]: I took over that terminology from Ankur (and believe I also heard that before myself, hadn't use the term "replication log" in that context though).

Just double-check with ChatGPT and it seems to think the same:

image

Is ChatGPT trustworthy enough for you here or should we check further? 🕵️

File: content/400-pulse/100-what-is-pulse.mdx (hunk 9 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx:93]: > I really wish GitHub would allow me to comment on lines not changed.

Been there, that's super annoying indeed 🙃 

> could we rename the header "Examples" to "Example projects", to be more specific? I'd be more excited to look at example projects than "examples", where maybe I find snippets, maybe just text, who knows, but "Example projects" is definitely code, and more exciting.

Agree and done!

File: content/400-pulse/200-getting-started.mdx (hunk 13 above)
Comment (nikolasburk) [reply content/400-pulse/200-getting-started.mdx]: I was referring to this part: 

image

It's the "**Automatic setup** for **Database replication**". We can link to "Logical replication" under "Concepts": https://www.prisma.io/docs/pulse/concepts#logical-replication 

```suggestion
- decide whether you want to use the **Automatic setup** for [**Database replication**](/pulse/concepts#logical-replication) (only available on paid plans)
```

I'm actually not a fan of this "Concepts" page and may want to change it in the future as well but for now this seems like the best location to link to on this topic?

File: content/400-pulse/200-getting-started.mdx (hunk 14 above)
Comment (nikolasburk) [reply content/400-pulse/200-getting-started.mdx:44]: Not sure I follow exactly!

These are the "Getting started" instructions, which assume that people don't have an API key (for that particular environment) yet. If they had one, I believe they wouldn't be able to go through the previous steps where they set up Pulse (e.g. click the **Enable Pulse** button because once Pulse is enabled, this button says **Manage Pulse**).

When you set up Pulse for the first time using the instructions above, you do need to explicitly generate that API key. Just double-checked that workflow:

image

File: content/400-pulse/200-getting-started.mdx (hunk 15 above)
Comment (nikolasburk) [reply content/400-pulse/200-getting-started.mdx:102]: yup, added the annotation to the code block to highlight the subscription lines: `highlight=9-13;add`

File: content/400-pulse/100-what-is-pulse.mdx (hunk 1 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx]: Good point, rephrased the intro like so:

[Prisma Pulse](https://www.prisma.io/data-platform/pulse) facilitates the implementation of event-driven workflows and architectures by providing managed infrastructure that captures databases events and distributes them to your applications. 

It lets you subscribe to any changes happening in your database using [Prisma Client](/orm/prisma-client) to power **real-time use cases** in your applications, such as:

File: content/400-pulse/100-what-is-pulse.mdx (hunk 5 above)
Comment (petradonka) [reply content/400-pulse/100-what-is-pulse.mdx]: I'd drop social interaction, or maybe rephrase to a more generic "updating user interfaces for changes" (or a smarter phrasing).

Real-time collaboration could then be it's own bullet.

And add "many" to "... and many other functionality"

File: content/400-pulse/100-what-is-pulse.mdx (hunk 6 above)
Comment (petradonka) [reply content/400-pulse/100-what-is-pulse.mdx]: > But I see how this can be turned into text-based content where essentially only the illustration of step 2. is needed because everything else is text and code snippets?
I would probably still not use a bullet list but rather use numbered headings like:

Yep, you get it!

I personally would prefer it, because I think it's more readable, more easily updateable, and can still well integrate smaller visuals. But up to you if you feel strongly about it.

File: content/400-pulse/200-getting-started.mdx (hunk 13 above)
Comment (petradonka) [reply content/400-pulse/200-getting-started.mdx]: Sure, sounds good!

File: content/400-pulse/200-getting-started.mdx (hunk 14 above)
Comment (petradonka) [reply content/400-pulse/200-getting-started.mdx:44]: So, you could skip this if you already have an API key. Should we mention that?

Comment (nikolasburk) [reply content/400-pulse/200-getting-started.mdx:44]: Added this note:

image

File: content/400-pulse/100-what-is-pulse.mdx [hunk 18]
```diff
@@ -6,29 +6,55 @@ tocDepth: 3
 toc: true
 ---
 
-<TopBlock>
+## What is Prisma Pulse?
```
Comment (nikolasburk) [inline content/400-pulse/100-what-is-pulse.mdx]:
```suggestion
## Overview
```

File: content/400-pulse/100-what-is-pulse.mdx (hunk 4 above)
Comment (jharrell) [reply content/400-pulse/100-what-is-pulse.mdx]: I'll trust it 👍

File: content/400-pulse/100-what-is-pulse.mdx (hunk 3 above)
Comment (jharrell) [reply content/400-pulse/100-what-is-pulse.mdx]: I think what you wrote makes sense, but maybe this would be better? I'm not tied to it, so definitely not a required change.
```suggestion
Instead, Prisma Pulse is based on the idea of _unidirectional data flow_ which is implemented via [Change Data Capture](https://en.wikipedia.org/wiki/Change_data_capture) (CDC). CDC gets rid of the deficiencies of the previously mentioned approaches and solves the problem of reacting to database events in a robust and elegant manner.
```

File: content/400-pulse/100-what-is-pulse.mdx (hunk 6 above)
Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx]: I went with that approach for now because I don't have a vertical layout yet: https://13a1647e.docs-51g.pages.dev/pulse/what-is-pulse

image

I think it works but what's a bit weird is that the illustration is so huge compared to the text.

Comment (nikolasburk) [reply content/400-pulse/100-what-is-pulse.mdx]: Got it fixed with Alejandro: 

image

I think the PR should be ready to merge now!

**Review by jharrell — APPROVED** [review]

**Review by petradonka — APPROVED** [review]
